src/http/websocket_http_client.zig

```diff
@@ -1320,23 +1320,17 @@ pub fn NewWebSocketClient(comptime ssl: bool) type {
                     .extended_payload_length_64, .extended_payload_length_16 => |rc| {
                         const byte_size = switch (rc) {
                             .extended_payload_length_64 => @as(usize, 8),
-                            .extended_payload_length_16 => @as(usize, 2),
+                            .extended_payload_length_16 => @min(@as(usize, 2), data.len),
                             else => unreachable,
                         };
                         // we need to wait for more data
                         if (data.len == 0) return;
 
-                        if (data.len < byte_size) {
-                            this.terminate(ErrorCode.control_frame_is_fragmented);
```

Comment:
why is this code being removed? If data.len = 3 but byte_size = 8, bad things will surely happen in line 1337/1331.

Reply:
we will need here a way to buffer the control frame instead of just hiting .need_body

Reply:
There's a comment stating "Control frames must not be fragmented", but it seems in the OP's case, they sometimes are, and if node's `ws` handles that we should too. Is that the problem?

Reply:
Exactly